exectors/nodejs/numberformat.js

```diff
@@ -0,0 +1,76 @@
+// Tests NumberFormat and DecimalFormat classes
+
+  let patternsToOptions = new Map();
+  patternsToOptions.set("0.0", {minimumFractionDigits: 1});
+  patternsToOptions.set("00", {minimumSignificantDigits: 1});
+  patternsToOptions.set("0.0", {minimumFractionDigits: 1});
+  patternsToOptions.set("@@@", {minimumSignificantDigits: 3});
+  patternsToOptions.set("@@###", {minimumSignificantDigits: 2,
+                                  maximumSignificantDigits: 5});
+  patternsToOptions.set("0.0000E0", {notation: "scientific",
+                                     minimunFractionDigits: 4});
+
+module.exports = {
+  decimalPatternToOptions: function(pattern, rounding) {
+    // TODO: Fill in with options from
+    let options = {};
+    if (patternsToOptions.has(pattern)) {
+      options = patternsToOptions.get(pattern);
+    }
+    if (rounding) {
+      options['roundingMode'] = rounding;
+    }
+    return options;
+  },
+
+  testDecimalFormat: function(json) {
+    const label = json['label'];
+
+    const pattern = json['pattern'];
+    const rounding = json['rounding'];
+    const input = parseFloat(json['input']);
+
+    let options;
+    let error = "unimplemented pattern";
+
+    // If options are in the JSON, use them...
+    try {
```

Comment:
Is there another way to find out if 'options' are available than try/catch? try/catch should be used mainly for error handling but not for testing for parameter availability.

Reply:
Yes, fixed this.